### DISCUSSION TITLE
HttpTransfer with upload and download at the same time?

### DISCUSSION THREAD

**Discussion by RobGibbens** [shinyorg/shiny#1489] [Q&A] [answered]
Is it possible to use background HttpTransfers to POST some json to an endpoint (effectively an upload) and still download the resulting json (either as json, or saved to a file)? It appears that HttpTransfers only support one or the other?

**Comment by aritchie** [comment]
So the HTTP Transfers feature is generally meant for blobs, so you should only expect some form of large data form back that saves to a file.  

That being said, you can post a header size chunk of JSON to the server with a transfer in any direction.

**Reply by RobGibbens** [reply]
Thanks @aritchie . So, in this scenario I’m making multiple POSTs to download JSON data in batches. This could take a long time, so I want to be be sure they don’t get interrupted when the app goes into the background. Would Shiny Jobs be more appropriate for this scenario?

**Reply by aritchie** [reply, marked as answer]
HTTP transfers is more appropriate for background data transfers.  There is unfortunately, no guarantee that it won't get interrupted.  iOS plays by whatever rules it wants.  The difference between jobs and transfers is that transfers won't give up trying.  iOS will time slice the transfer way better vs what it will give to a job.

The next fun part here is that if you need to process your JSON files, you won't get a ton of time after the transfer finishes to that, so you may have to partner a job to get the processing portion done.
